Added generic save shutdown method for RetroPie

crcerror opened the pull request:
Hello,

here is a generic method for save shutdown in every situation.
If you want you can use the added ./multi_switch.sh as standalone switch.
Just call it with `./multi_switch.sh --nespi+` to get working buttons with bash script only.

If you want to use python code you can use switches `--es-pid` and `--es-poweroff` to detect and save shutdown RetroPie!

All you need to code is change code to your installer script.

Greetings

crcerror committed:
Added generic shutdown method

Works only for RetroPie!
Please copy ./multi_switch.sh

crcerror committed:
Added generic safe shutdown (save metadata!)

Added generic method
Working on RetroPie 4.3 and 4.4
By cyperghost

crcerror committed:
v 0.41

Working perfect now!

crcerror committed:
Save Shutdown for RetroPie - never loose data

Cleaned code a bit
Used better logic

crcerror commented:
I changed the code logic a bit to use integer values
So boolean in if/else branch is working

crcerror committed:
Update SafeShutdown.py

crcerror committed:
More natural behaviour

1. If you press restart if emulator is currently running, then you will be kicked back to ES main menu
2. If you press restart in ES main screen, ES will be restartet (no reboot!)
3. If you press power-off then Raspberry will shutdown

All metadata is always saved
Best -> cyperghost

UncleRus commented:
Repo owners simply ignores issues and PRs.
I think that we need a fork.

crcerror committed:
Update README.md

crcerror committed:
Update install.sh

crcerror commented:
Not an official but works....
Improved with more natural behaviour
1. If an emulator is running inside ES and you press reset, you will be kicked back to ES
2. If you are in ES main screen and press reset, then ES will be restarted (NO REBOOT! - that makes simply no sense)
3. If you press power off .... well ... what do you expect?

All metadata is 100% be saved, smooth, fast, smart :)

crcerror committed:
Delete recalbox_install.sh

crcerror committed:
Delete recalbox_SafeShutdown.py

crcerror committed:
Update README.md

crcerror committed:
Update README.md

crcerror committed:
Update README.md

crcerror committed:
Support RPi 3B+ now

Removed UART
Caused some issues with Raspberry 3B+ devices

ddaversa commented:
Nice mods. If I may suggest, it would be cool if we could have the reset button open up the Retroarch menu when in-game, similar behavior to the Classic Series.

Commit:
Ignore if the directory already exists for install over RetroFlag version

crcerror committed:
Added mode for RPi 3B+

Disabeling UART now, this will prevent some issues
with CPU throtteling and makes use of advanced setup!

crcerror committed:
Merge pull request #1 from KlfJoat/bug-install-mkdir

Ignore if the directory already exists for install over RetroFlag version

crcerror committed:
v 0.85

Updated with latest multi_switch version

crcerror committed:
Updated installer

Introduced wget -N switch to overwrite old installed files - preventing failed installs

crcerror committed:
install.sh updated

Added messages for installing files 
now with progress bar
Write status of install to stdout

crcerror committed:
install.sh updated

Added messages for installing files 
now with progress bar
Write status of install to stdout
Removed typo

crcerror committed:
Create recalbox_SafeShutdown.py

crcerror committed:
Create recalbox_SafeShutdown.sh

crcerror committed:
Create recalbox_install.sh

crcerror committed:
Create install_batocera.sh

crcerror committed:
Update install_batocera.sh

crcerror committed:
Update install_batocera.sh

crcerror committed:
Update install_batocera.sh

crcerror committed:
Update install_batocera.sh

crcerror committed:
Update install_batocera.sh

crcerror committed:
Update install_batocera.sh

crcerror committed:
Finally done

crcerror committed:
Update install_batocera.sh

crcerror committed:
Update install_batocera.sh

crcerror committed:
Delete install_batocera.sh

crcerror committed:
Create install_batocera.sh

crcerror committed:
Update install_batocera.sh

crcerror committed:
Delete install_batocera.sh

crcerror committed:
Create install_recalbox.sh

crcerror committed:
Update install_recalbox.sh

crcerror committed:
Update install_recalbox.sh

crcerror committed:
Update install_recalbox.sh

crcerror committed:
Update recalbox_install.sh

crcerror committed:
Update recalbox_install.sh

crcerror committed:
Create install_batocera.sh

crcerror committed:
Update recalbox_install.sh

crcerror committed:
Update recalbox_install.sh

crcerror committed:
Create batocera_SafeShutdown.py

crcerror committed:
Delete batocera_SafeShutdown.py

crcerror committed:
Create batocera_SafeShutdown.py

crcerror committed:
Update recalbox_install.sh

crcerror committed:
fixed indentation

crcerror committed:
fixed indentation

crcerror committed:
Update install_recalbox.sh

crcerror committed:
added scriptDir argument

Make scripts more portable

crcerror committed:
Delete batocera_SafeShutdown.py

crcerror committed:
Updated installer script

Added autodetection of system!

crcerror committed:
Update install_recalbox.sh

crcerror committed:
Update install_batocera.sh

crcerror committed:
Fixed pathes to execute files

crcerror committed:
Fixed get several IPs

crcerror committed:
Final!

Fixed some logic errors!

crcerror committed:
Final!

Fixed typo!

crcerror committed:
Added --kodi switch

We can now start KODI with button center.
It will behave exactly as in the original system

crcerror committed:
Update README.md

crcerror committed:
Update README.md

crcerror committed:
Update README.md

crcerror committed:
Fix installation

Batocera System 5.21 is supported now!

crcerror committed:
Enabled UART again

This is needed for Recalbox and Batocera systems to make LED work proper - sadly!

crcerror committed:
Update recalbox_SafeShutdown.py

crcerror committed:
--shutdown parameter

crcerror committed:
Update recalbox_SafeShutdown.py

crcerror committed:
got back old version

crcerror committed:
Save Metadata in BATO 5.22 systems now

crcerror committed:
Save Metadata in BATOCERA 5.22 builds now

crcerror committed:
Save Metadata on BATOCERA 5.22 builds now

crcerror committed:
Added SIGTERM to some emulators

crcerror committed:
need BATCERA 5.24 and above

crcerror committed:
BATOCERA 5.24 needed at least

crcerror committed:
Create batocera_install.sh

crcerror committed:
Updated for BATOCERA 5.24

crcerror committed:
BATOCERA 5.24 needed for this

Resolved a small typo

crcerror committed:
BATOCERA 5.24 at least needed

configurated for 5.24
removed the misstype

crcerror committed:
BATOCERA 5.24 at least needed

removed some typos
version correction

crcerror committed:
BATOCERA 5.24 (dev 10102019) and beyond

crcerror committed:
Update README.md

crcerror committed:
Update README.md

crcerror committed:
BATOCERA 5.24 (dev 10102019) and beyond 

Added excutable flag for script!

crcerror committed:
py: BATOCERA 5.24 (dev 10102019) and beyond

Added SHEBANG for python script

crcerror committed:
BATOCERA 5.25 support added

crcerror committed:
Create install.sh

crcerror committed:
Create SafeShutdown.py

crcerror committed:
Update install.sh

crcerror committed:
Create batocera_install.sh

crcerror committed:
Create install_gpi.sh

crcerror committed:
Create install_gpi_batocera.sh

crcerror committed:
Update README.md

crcerror committed:
Create uninstall_all.sh

crcerror committed:
Update for GPi

crcerror committed:
Added uninstaller

crcerror committed:
BATOCERA: fixed installer GPi

NetizenAbel committed:
small fixes to README

Spelling and formatting fixes

crcerror committed:
Merge pull request #3 from kashaiahyah85/patch-1

small fixes to README

DeeKay-Deluxe committed:
Fixed install.sh

Fixed the link to the SafeShutdown.py script in line 29.

crcerror committed:
Merge pull request #4 from DeeKay-Deluxe/master

Fixed install.sh

crcerror committed:
Support for NESPi4 added

crcerror committed:
Added NESPi4 support

crcerror committed:
Added NESPi4 support

crcerror committed:
NESPi4 support added

crcerror committed:
Added NESPi4 support

crcerror committed:
Added best NESPi4 support

crcerror committed:
Added retroflag NESPi4 removal

crcerror committed:
Update batocera_install.sh

cgherman commented:
For those still looking here, fork was created at https://github.com/crcerror/retroflag-picase/

crcerror committed:
Added python3 and python removal method
